fe/estimator_abfe.py

```diff
@@ -300,51 +240,57 @@ def _deltaG(model, sys_params) -> Tuple[Tuple[float, List], np.array]:
         for future in futures:
             results.append(future.result())
 
-    mean_du_dls = []
     all_grads = []
 
     if model.endpoint_correct:
-        ti_results = results[:-1]
+        sim_results = results[:-1]
     else:
-        ti_results = results
+        sim_results = results
 
-    for lambda_idx, (lambda_window, result) in enumerate(zip(model.lambda_schedule, ti_results)):
-        # (ytz): figure out what to do with stddev(du_dl) later
-        # print(f"{model.prefix} index {lambda_idx} lambda {lambda_window:.5f} <du/dl> {np.mean(result.du_dls):.5f} med(du/dl) {np.median(result.du_dls):.5f}  o(du/dl) {np.std(result.du_dls):.5f}")
-        mean_du_dls.append(np.mean(result.du_dls))
+    U_kn = []
+    N_ks = []
+    for lambda_idx, (lambda_window, result) in enumerate(zip(model.lambda_schedule, sim_results)):
+        U_kn.append(result.lambda_us)
+        N_ks.append(len(result.lambda_us))
         all_grads.append(result.du_dps)
 
-    tibar_dG = 0
+    K = len(model.lambda_schedule)
+    U_kn = np.array(U_kn) # [K, F, K]
+    U_kn = np.reshape(U_kn, (-1, K)).transpose() # [K, F*K]
+    # deal with overflow issues here, tbd: make it more robust?
+    U_kn = np.where(U_kn < -1e10, np.inf, U_kn)
```

Comment:
Some near-duplication is arising in places where we need to detect overflow / replace suspicious negative numbers with +inf. May want to consolidate these once we settle on a threshold to use across the project.

Reply:
agreed. energies are actually quite tricky because the threshold is going to be size-extensive.